File: OOP/2_method.cpp
#include <iostream>
using namespace std;

class A {
    // No members defined
};

// Define a function outside of the class
void display() {
    cout << "I am Methods" << endl;
}

int main() {
    A obj; // Create an instance of class A

    // Call the display function
    display(); // Function call

    // Output the address of the function (similar to printing a reference in Python)
    cout << (void*)display << endl; // Print the function pointer
    cout << (void*)display << endl;
    // Print the object representation (default behavior)
    cout << &obj << endl; // Print the address of the object

    return 0;
}

File: OOP/2_method.py

class A:
    pass

def display():   # without self is not a method of class A
    print("I am Methods")


obj = A()

obj.value = display  # obj.value holo display function er reference
obj.value()  # function ke call kora holo
print(obj.value)  # reference function er address print hobe
print(obj)  #object print hobe



# obj.display()    #In Python, we can only call methods that belong to a class from an instance of that class. Since display() is defined outside the class, the object obj doesn't know about it.it raise a error



